core/src/display_object/movie_clip.rs

```diff
@@ -163,6 +163,9 @@ pub struct MovieClipData<'gc> {
     has_focus: bool,
     avm2_enabled: bool,
 
+    /// The cached AVM2 class AVM1Movie corresponding to this MovieClip.
+    avm2_avm1movie: Option<Avm2Object<'gc>>,
```

Comment:
also general question: should we maybe drop the `AvmObject` enum here and just split it into
```
// avm1 MC
object1: Option<Avm1Object<'gc>>,
// avm2 MC or AVM1Movie
object2: Option<Avm2Object<'gc>>,
```
?